core/src/tests/dummy_transport.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2018 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+//! `DummyTransport` is a `Transport` used in tests. It implements a bare-bones
+//! version of the trait along with a way to setup the transport listeners with
+//! an initial state to facilitate testing.
+
+use futures::prelude::*;
+use futures::{future::{self, FutureResult}, stream};
+use {Multiaddr, Transport};
+use std::io;
+
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub(crate) enum ListenerState {
+	Ok(Async<Option<usize>>),
```

Comment:
The idea here was just to put "something" in there, preferably non-void. I see why it could be surprising to the reader; what's a better choice? I did add a comment though?

Reply:
Initially I assumed it would be a dummy, in which case I would have suggested `()` instead. However one can write tests which check the item index of a listener (hence my other suggestion (`stream::iter_ok(n..)`). Consequently I would write a comment stating that the usize value indexes items produced by the listener.